handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -237,36 +237,38 @@ protected void deallocate() {
         keyMaterialManager = context.keyMaterialManager();
         enableOcsp = context.enableOcsp;
 
-        Lock readerLock = context.ctxLock.readLock();
-        readerLock.lock();
-        try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
-        } finally {
-            readerLock.unlock();
-        }
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
+            Lock readerLock = context.ctxLock.readLock();
+            readerLock.lock();
+            try {
+                ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
> If there were concurrent ReferenceCountedOpenSslEngine initializations

I'm not sure if this is what you mean but generally a single `SslContext` is shared across multiple threads (e.g. `EventLoop`s) to create new `SslEngine`s. This was the motivating factor for using the reader/writer lock on the `SslContext` access instead of a `synchronized` block.